### PR TITLE
fix(eslint-plugin): add TSPropertySignature with TSFunctionType annotation to typeMethod selector

### LINKED ISSUE #6617: Bug: [naming-convention] `typeMethod` selector doesn't support function expressions
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBACgTgezAZygXigbwFBSgenygGME44JjgoA7AQwFsJlcoEaAxBBACgEoAXFABuCAJYATANyt2AITpwh-dAD4R46dlaEoYmqXKVq9JizwALOjQkAbCF16CNkmZet2ICpVBVp1oq7YAL5SQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6JgQwFtmBzWsgHsmAN0QtKw9GADa4bDkTRog6JAA08hVgXZIyREjL5V0yEVIBZRPgAWggCYatuyMWiIAZpQAe0ubquUi4KALqagTieqlwc+P6QAMr4VMYAChzIZBzwAMKZiJChIWAAvi7FEOWlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
type Props = {
  // correct names
  onFoo(): void;
  onBar: () => void;

  // incorrect names
  handleFoo(): void;
  handleBar: () => void;
};
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "selector": "typeMethod",
        "prefix": [
          "on"
        ],
        "format": ["StrictPascalCase"]
      }
    ]
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Both `handleFoo` and `handleBar` should be marked as errors.

### Actual Result

Only `handleFoo` is marked as an error. 

### Additional Info

The `typeMethod` selector should pick up every possible method syntax similarly to the `objectLiteralMethod` selector. Right now only `TSMethodSignature` nodes are picked up by the `typeMethod` selector, whereas it should also pickup `TSPropertySignature` nodes with `typeAnnotation` of `TSFunctionType`. 

If you confirm this is a bug, I would be more than happy to create a PR with the fix 😄 

References in code:
- place where every non-computed `TSPropertySignature` is mapped as `typeProperty` selector: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/naming-convention.ts#L397
- place where every non-computed `typeMethod` selector is mapped: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/src/rules/naming-convention.ts#L463

### PR DESCRIPTION
**Pull request by kozlovvski**
## PR Checklist

- [x] Addresses an existing open issue: fixes #6617 
- [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
- [x] Steps in [Contributing](https://typescript-eslint.io/contributing) were taken

## Overview

[Playground like with the reproduces issue](https://typescript-eslint.io/play/#ts=4.9.3&sourceType=module&code=C4TwDgpgBACgTgezAZygXigbwFBSgenygGME44JjgoA7AQwFsJlcoEaAxBBACgEoAXFABuCAJYATANyt2AITpwh-dAD4R46dlaEoYmqXKVq9JizwALOjQkAbCF16CNkmZet2ICpVBVp1oq7YAL5SQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6JgQwFtmBzWsgHsmAN0QtKw9GADa4bDkTRog6JAA08hVgXZIyREjL5V0yEVIBZRPgAWggCYatuyMWiIAZpQAe0ubquUi4KALqagTieqlwc+P6QAMr4VMYAChzIZBzwAMKZiJChIWAAvi7FEOWlQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA)

**Issue summary**
Currently the `typeMethod` selector in `naming-convention` rule options isn't applied to `foo: () => void` syntax. Instead it is caught by `typeProperty` selector.

**What is changed**
Only the `naming-convention.ts` file + related tests.

**How the code works**:
Instead of treating every `TSPropertySignature` as `typeProperty` selector, nodes with `typeAnnotation.typeAnnotation.type` equaling `TSFunctionType` are matched by `typeMethod` selector.

### CODE REVIEW

File: packages/eslint-plugin/src/rules/naming-convention.ts [hunk 1]
```diff
@@ -394,20 +394,21 @@ export default util.createRule<Options, MessageIds>({
...
-        validator: validators.typeProperty,
-        handler: (
-          node: TSESTree.TSPropertySignatureNonComputedName,
-          validator,
-        ): void => {
-          const modifiers = new Set<Modifiers>([Modifiers.public]);
-          if (node.readonly) {
-            modifiers.add(Modifiers.readonly);
-          }
+      'TSPropertySignature[computed = false][typeAnnotation.typeAnnotation.type != "TSFunctionType"]':
```
Comment (kozlovvski) [inline packages/eslint-plugin/src/rules/naming-convention.ts:397]: I'm wondering if this `typeAnnotation.typeAnnotation.type` part should be moved to some variable. It is later on used in L466 in `typeMethod` selector

Comment (JoshuaKGoldberg) [reply packages/eslint-plugin/src/rules/naming-convention.ts:397]: You know, we've been leaning away from using these fancier queries, and instead towards simpler, more readable ones. https://github.com/eslint-community/eslint-plugin-eslint-plugin/issues/339

Not requesting changes here as we haven't taken action on that yet & this file already has a lot of complex queries. Just food for thought. 😄

**Review by JoshuaKGoldberg — APPROVED** [review]
Thus the complexity of `naming-convention.ts` increases its perpetual march towards infinity.

Thanks for this @kozlovvski, looks good! ✨

File: packages/eslint-plugin/src/rules/naming-convention.ts [hunk 2]
```diff
@@ -394,20 +394,21 @@ export default util.createRule<Options, MessageIds>({
...
+        {
+          validator: validators.typeProperty,
+          handler: (
+            node: TSESTree.TSPropertySignatureNonComputedName,
+            validator,
+          ): void => {
+            const modifiers = new Set<Modifiers>([Modifiers.public]);
+            if (node.readonly) {
+              modifiers.add(Modifiers.readonly);
```
Comment (JoshuaKGoldberg) [inline packages/eslint-plugin/src/rules/naming-convention.ts:406]: Oh sorry @kozlovvski - the codecov app had a delay today and I missed this. Could you please add test cases to exercise this line? It's not covered right now.

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Whoops, sorry for the fakeout - one untested line away from approval!

File: packages/eslint-plugin/src/rules/naming-convention.ts (hunk 2 above)
Comment (kozlovvski) [reply packages/eslint-plugin/src/rules/naming-convention.ts:406]: @JoshuaKGoldberg I can try, but I am not sure what this part does. I didn't adjust the `handler` -> it's marked as changed due to indentation

Comment (kozlovvski) [reply packages/eslint-plugin/src/rules/naming-convention.ts:406]: @JoshuaKGoldberg okay, I think it should be fixed now 🙌🏻

**Review by JoshuaKGoldberg — APPROVED** [review]
💯 thanks!

**Pull request merged**